### PR TITLE
Feature/bb 52

### PR DESCRIPTION
**Pull request by YanaHV**

### CODE REVIEW

**Review by OleksandrKrasilovskyi — DISMISSED** [review]

File: src/main/java/com/barterbay/app/domain/dto/AddressDTO.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  private Long id;
+  @NotBlank(message = "Country cannot be empty")
+  private String country;
+  @NotBlank(message = "City cannot be empty")
+  private String city;
+  @NotBlank(message = "Street cannot be empty")
+  private String street;
+  @NotBlank(message = "ZIP code cannot be empty")
+  private int zipCode;
+  private GoodDTO goodDTO;
```
Comment (kolirnicolas) [inline src/main/java/com/barterbay/app/domain/dto/AddressDTO.java:20]: why do you need good DTO inside Adress DTO

File: src/main/java/com/barterbay/app/domain/dto/GoodDTO.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+public class GoodDTO {
+  private Long id;
+  @NotBlank(message = "Good`s name cannot be empty")
+  private String goodName;
+  @NotBlank(message = "Description cannot be empty")
+  private String description;
+  @NotBlank(message = "Category cannot be empty")
+  private Category category;
+  private AddressDTO address;
+  private UserDTO userOwner;
```
Comment (kolirnicolas) [inline src/main/java/com/barterbay/app/domain/dto/GoodDTO.java:20]: Please use long insted of object

`private Long userOwnerId;`

Comment (OleksandrKrasilovskyi) [reply src/main/java/com/barterbay/app/domain/dto/GoodDTO.java:20]: I agree

File: src/main/java/com/barterbay/app/mapper/AddressMapper.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+import com.barterbay.app.domain.Address;
+import com.barterbay.app.domain.dto.AddressDTO;
+import org.mapstruct.Mapper;
+
+@Mapper(componentModel = "spring")
+public interface AddressMapper {
+
+  Address addressDTOToAddressEntity(AddressDTO addressDTO);
+
```
Comment (OleksandrKrasilovskyi) [inline src/main/java/com/barterbay/app/mapper/AddressMapper.java:11]: Excessive line

File: src/main/java/com/barterbay/app/domain/Good.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  private Category category;
+
+  @OneToOne(fetch = FetchType.LAZY)
+  @JoinColumn(name = "address_id", nullable = false)
+  private Address address;
+
+  @ManyToOne(fetch = FetchType.LAZY)
+  @JoinColumn(name = "user_id", nullable = false)
+  private User user;
+
```
Comment (OleksandrKrasilovskyi) [inline src/main/java/com/barterbay/app/domain/Good.java:57]: Excessive line

File: src/main/java/com/barterbay/app/domain/dto/AddressDTO.java (hunk 1 above)
Comment (YanaHV) [reply src/main/java/com/barterbay/app/domain/dto/AddressDTO.java:20]: corrected

File: src/main/java/com/barterbay/app/domain/dto/GoodDTO.java (hunk 2 above)
Comment (YanaHV) [reply src/main/java/com/barterbay/app/domain/dto/GoodDTO.java:20]: corrected

File: src/main/java/com/barterbay/app/domain/dto/GoodDTO.java [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+import com.barterbay.app.enumeration.Category;
+import lombok.Getter;
+import lombok.Setter;
+
+import javax.validation.constraints.NotBlank;
+
+@Getter
+@Setter
+public class GoodDTO {
```
Comment (OleksandrKrasilovskyi) [inline src/main/java/com/barterbay/app/domain/dto/GoodDTO.java:11]: How do you use this dto?
What is this for?

**Pull request closed without merging**